### PR TITLE
Coverage for genesis package

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added extra coverage on genesis package

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: genesis/process/genesisBlockCreator.go [hunk 1]
```diff
@@ -167,18 +167,12 @@ func checkArgumentsForBlockCreator(arg ArgsGenesisBlockCreator) error {
 	if check.IfNil(arg.Data.StorageService()) {
 		return process.ErrNilStore
 	}
-	if check.IfNil(arg.Data.Blockchain()) {
```
Comment (ssd04) [inline genesis/process/genesisBlockCreator.go:170]: :+1:

**Review by ssd04 — APPROVED** [review]

File: genesis/parsing/accountsParser_test.go [hunk 2]
```diff
@@ -744,3 +747,11 @@ func TestAccountsParser_GenerateInitialTransactionsVerifyTxsHashes(t *testing.T)
 		assert.Equal(t, tx, v.Transaction)
 	}
 }
+
+func TestAccountsParser_GenesisMintingAddress(t *testing.T) {
+	t.Parallel()
+
+	ap := parsing.NewTestAccountsParser(createMockHexPubkeyConverter())
+	addr := ap.GenesisMintingAddress()
+	assert.Equal(t, hex.EncodeToString([]byte("erd17rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rcqqkhty3")), addr)
```
Comment (ssd04) [inline genesis/parsing/accountsParser_test.go:756]: maybe extract minting address into a const since we have it in 3 places in this package

Comment (bogdan-rosianu) [reply genesis/parsing/accountsParser_test.go:756]: []byte("erd1...") is confusing. maybe use the real bech32 converter or use a hex / dummy address here

File: genesis/process/genesisBlockCreator_test.go [hunk 3]
```diff
@@ -222,6 +249,213 @@ func createMockArgument(
...
+		arg.GasSchedule = nil
+
+		gbc, err := NewGenesisBlockCreator(arg)
+		require.True(t, errors.Is(err, process.ErrNilGasSchedule))
+		require.Nil(t, gbc)
+	})
+	t.Run("nil GasSchedule should error", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockArgument(t, "testdata/genesisTest1.json", &mock.InitialNodesHandlerStub{}, big.NewInt(22000))
```
Comment (bogdan-rosianu) [inline genesis/process/genesisBlockCreator_test.go:402]: duplicated test

**Review by bogdan-rosianu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            genesis_tests                                ##
================================================================================
- Known Warnings              8
- New Warnings                1
- Known Errors                1
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  10909
- wrong nonce in block                       3759
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the testnet
================================================================================

**Pull request merged**